app/services/alerts/generate_and_save_alerts.rb

```diff
@@ -2,39 +2,83 @@ module Alerts
   class GenerateAndSaveAlerts
     def initialize(school:, framework_adapter: FrameworkAdapter, aggregate_school: AggregateSchoolService.new(school).aggregate_school)
       @school = school
-      @alert_framework_adapter = framework_adapter
+      @alert_framework_adapter_class = framework_adapter
       @aggregate_school = aggregate_school
+      @alert_generation_run = AlertGenerationRun.create(school: @school)
+      @alerts_to_save = []
+      @alert_errors_to_save = []
```

Comment:
Not sure these need to be instance variables? Could perform also collect them?

Possibly unlikely to happen, but what would the intended effect of running `perform` twice be? Currently it would produce a single generation run with 3 lots of alerts/errors in. One lot from the first run, and then the lot from the first run again and the lot added from the second run.